Proposal bind shorthand

cesarParra opened the pull request:
Bind feature based on this proposal: https://github.com/cesarParra/lwc-signals/issues/38

cesarParra committed:
Proposal - bind shorthand

cesarParra committed:
Merge branch 'main' into proposal-bind-shorthand

# Conflicts:
#	force-app/lwc/signals/core.js
#	src/lwc/signals/core.ts

cesarParra committed:
Merging latest from main

github-actions[bot] committed:
Commit from GitHub Actions (CI)

cesarParra commented on src/lwc/signals/core.ts:
```diff
@@ -579,4 +579,20 @@ function isSignal(anything: unknown): anything is Signal<unknown> {
+  }
+}
+
+function $binded(component: Record<string, object>, propertyName: string) {
```
Rename to just bind (or bound)

stanleyzapata replied:
go with bind it's pretty standard

stanleyzapata commented on examples/counter/lwc/countTracker/countTracker.js:
```diff
@@ -1,8 +1,10 @@
 import { counter, counterPlusTwo } from "c/demoSignals";
 
 export default class CountTracker extends LightningElement {
+  bindTest = $binded(this, "bindTest").to(counter);
```
I understand why you are doing this but I don't love it. Not sure what would be a better way to do the binding

cesarParra replied:
Yeah, that's why I'm keeping it as a proposal stage until I get some feedback, not sure if you saw the issue I put up for discussion: https://github.com/cesarParra/lwc-signals/issues/38

I don't want to merge if its going to be detrimental compare to what exists today, but I haven't found a better way of doing it with how Javascript works today. The only way I can think this could be done is by using [decorators](https://github.com/tc39/proposal-decorators), but JS doesn't support those yet.

stanleyzapata commented on force-app/lwc/signals/core.js:
```diff
@@ -348,4 +348,20 @@ function $resource(fn, source, options) {
+    this.propertyName = propertyName;
+  }
+  to(signal) {
+    $effect(() => {
```
is this ok performance wise?

stanleyzapata committed:
Allows circular dependencies in effects with a stack depth limit

stanleyzapata committed:
Renamed variable Names

stanleyzapata committed:
Adds explanation for MAX_STACK_DEPTH constant

stanleyzapata committed:
feat: circular dependecy management logic update

- Introduced a `stackDepth` tracker in effect execution to manage circular dependencies.
- Effects can now re-enter up to a maximum stack depth of 16 without throwing an error.

This change allows limited circular dependencies while preventing infinite loops.

semantic-release-bot committed:
chore(release): Version 1.4.0 - Automated release [skip ci]

See the release details: v1.4.0

stanleyzapata committed:
fix: Build Last Change (#40)

* fix: Rebuild

* Commit from GitHub Actions (CI)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

semantic-release-bot committed:
chore(release): Version 1.4.1 - Automated release [skip ci]

See the release details: v1.4.1

cesarParra committed:
Merge remote-tracking branch 'origin/proposal-bind-shorthand'

github-actions[bot] committed:
Commit from GitHub Actions (CI)

cesarParra committed:
Bind decorator

github-actions[bot] committed:
Commit from GitHub Actions (CI)

cesarParra committed:
Doc update

github-actions[bot] committed:
Commit from GitHub Actions (CI)
